azavea/pwd-unitybar
# Move React & React-DOM to peer-dependencies

## Overview

While working on trying to reduce the GARP application bundle size, I noticed that the Unity Bar was pulling in a full copy of React-DOM along with its own source code. Moving React & React-DOM to `peerDependencies` here will keep that from happening & ensure that the Unity Bar will just use bundle's existing React-DOM.

To allow moving those dependency declarations, the change also adjusts some eslint rules related to imports. I also added Webpack Bundle Analyzer for dev and production builds. In development, it'll run as a server on port 7778; when running `cibuild`, it will generate a .gitignored `report.html` file in the `dist/` dir.

Connects #52

### Demo

<img width="1248" alt="screen shot 2019-01-04 at 1 36 08 pm" src="https://user-images.githubusercontent.com/4165523/50704534-b5b6c800-1025-11e9-8176-fd36ebff62b3.png">

## Testing Instructions

- get this branch, then `npm install`
- `./scripts/server` then verify that the dev application launches on port 7777 and the bundle analyzer server launches on port 7778 (also check that the application still works as expected)
- `./scripts/cibuild` then verify that the process outputs production bundles and a `dist/report.html` file
- open `dist/report.html` in the browser and verify that you see the composition depicted above
